types/react-select/lib/components/index.d.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  ComponentType,
+  ReactElement as Element,
+} from 'react';
+import {
+  IndicatorContainerProps,
+  ContainerProps,
+  ValueContainerProps,
+  IndicatorsContainer,
+  SelectContainer,
+  ValueContainer,
+} from './containers';
+import {
+  IndicatorProps,
+  LoadingIconProps,
+  ClearIndicator,
+  DropdownIndicator,
+  LoadingIndicator,
+  IndicatorSeparator,
+  DownChevron,
+  CrossIcon,
+} from './indicators';
+
+import Control, { ControlProps } from './Control';
+import Group, { GroupProps, GroupHeading } from './Group';
+import Input, { InputProps } from './Input';
+import Menu, {
+  MenuProps,
+  MenuList,
+  MenuListComponentProps,
+  MenuPortal,
+  MenuPortalProps,
+  NoticeProps,
+  NoOptionsMessage,
+  LoadingMessage,
+} from './Menu';
+import MultiValue, {
+  MultiValueProps,
+  MultiValueContainer,
+  MultiValueLabel,
+  MultiValueRemove,
+} from './MultiValue';
+import Option, { OptionProps } from './Option';
+import Placeholder, { PlaceholderProps } from './Placeholder';
+import SingleValue, { SingleValueProps } from './SingleValue';
+
+export type PlaceholderOrValue =
+  | Element<ComponentType<PlaceholderProps>>
+  | Element<ComponentType<SingleValueProps>>
+  | Array<Element<ComponentType<MultiValueProps>>>;
+
+export type IndicatorComponentType = ComponentType<IndicatorProps>;
+
+export type SelectComponents = {
+  ClearIndicator: IndicatorComponentType | null,
+  Control: ComponentType<ControlProps>,
+  DropdownIndicator: IndicatorComponentType | null,
+  DownChevron: ComponentType<any>,
+  CrossIcon: ComponentType<any>,
+  Group: ComponentType<GroupProps>,
+  GroupHeading: ComponentType<any>,
+  IndicatorsContainer: ComponentType<IndicatorContainerProps>,
+  IndicatorSeparator: IndicatorComponentType | null,
+  Input: ComponentType<InputProps>,
+  LoadingIndicator: ComponentType<LoadingIconProps> | null,
+  Menu: ComponentType<MenuProps>,
+  MenuList: ComponentType<MenuListComponentProps>,
+  MenuPortal: ComponentType<MenuPortalProps>,
+  LoadingMessage: ComponentType<NoticeProps>,
+  NoOptionsMessage: ComponentType<NoticeProps>,
+  MultiValue: ComponentType<MultiValueProps>,
+  MultiValueContainer: ComponentType<any>,
+  MultiValueLabel: ComponentType<any>,
+  MultiValueRemove: ComponentType<any>,
+  Option: ComponentType<OptionProps>,
+  Placeholder: ComponentType<PlaceholderProps>,
+  SelectContainer: ComponentType<ContainerProps>,
+  SingleValue: ComponentType<SingleValueProps>,
+  ValueContainer: ComponentType<ValueContainerProps>,
+};
+
+export type SelectComponentsConfig = {
```

Comment:
Is this different from `SelectComponents` type? Are both needed?

Reply:
In original Flow typings it's made by `$shape`. I guess best way to put it in typescript would be `Partial<SelectComponents>`

Reply:
Thanks for the pointer, @burtek! I'll look into this.

Reply:
d